### ISSUE TITLE
More attention to mobile docks tagging

### ISSUE THREAD

**Issue by irvnriir** [openstreetmap/id-tagging-schema#120]
("Floating Dry docks" is usually a pontoon boats)

Mobile docks can be and are relocated relatively fast, making the maps outdated .

I suggest to change the "Wet Dock / Dry Dock" name to "Dock" which is more directly comprises a fixed structure (and basically more correct) .

Floating docks can/should be added with Tag:waterway=dock; , Tag:dock=floating; . But i see many marked wrongly as only Tag:waterway=dock; .

The issue is resulting from openstreetmap/iD#3030 which was solved by returning Pier with "Dock" search .

**Comment by tyrasd** [comment]
Can you please reiterate what the issue with the current preset is and what exactly you are proposing to be changed?

From what I read in https://github.com/openstreetmap/iD/issues/3030 is that the term _Dock_ does mean _Pier_ in American English, thus there was confusion among American mappers. This was solved by renaming the preset to _Wet Dock / Dry Dock_.

I think we could add presets for more specific dock types (e.g. _Floating Dock_, _Dry Dock_, _Tidal Dock_, …) to make it easier to tag those. But we would still need to keep around the generic preset for "unspecified" docks.

**Comment by irvnriir** [comment]
@tyrasd

> the term Dock (also) means Pier in American English, thus there was confusion among American mappers

\>-> but also it
> was solved by returning Pier (as result) with (from) "Dock" search

I guess you suggest to make it `Tag:waterway=floating-dock;` ? I don't know whether its a correct solution in regard to procedural processing of the data .

**Comment by irvnriir** [comment]
Or, can visible "Floating Dock" be added for `Tag:waterway=dock; Tag:dock=floating;`

**Comment by tyrasd** [comment]
> but also […] it was solved by returning Pier (as result) with (from) "Dock" search

Yes, but I was under the impression that the two changes in https://github.com/openstreetmap/iD/commit/f69714f83cc6c27a72924ab8a84f55c58b6f8d6d are supposed to work "together" to best avoid the confusion between piers and docks in American English. But since I'm not a native speaker, my interpretation shouldn't really count here.

> Or, can visible "Floating Dock" be added for Tag:waterway=dock; Tag:dock=floating; ?

Exactly this. The idea is to have:
* a "Floating Dock" preset which matches the combination of `waterway=dock` + `dock=floating`
* a "Dry Dock" preset which matches the combination of `waterway=dock` + `dock=drydock`
* perhaps also a "Tidal Dock" preset for `waterway=dock` + `dock=tidal`
* and a generic "Wet Dock / Dry Dock" preset, which matches all remaining `waterway=dock` features (e.g. ones without the `dock` tag present)

**Comment by irvnriir** [comment]
It seems that when the solutions was developed, devs didn't consider floating docks .

If "Floating dock" would be shown right next to "Dock", this would probably work .

I can't find specification of "Tidal Dock", either imagine a subject which would satisfy this name .

**Comment by irvnriir** [comment]
The definition at https://wiki.openstreetmap.org/wiki/Cs:Tag:waterway%3Ddock is rather of a "Harbor" .

**Comment by irvnriir** [comment]
@k-yle As i understand, the linked ["floating dock"](https://commons.wikimedia.org/wiki/File:Pasaia,_pantalana.jpg) is a "floating pier" in OSM, which does have the separation from "pier" in the search.

The issue is that ppl use the general "dock" without specification that its floating, which creates a severe incorrectness in the OSM data.

**Comment by k-yle** [comment]
correct, but that's true for many presets. Sometime we solve it like #1706, but here we have a major problem that `Pier` and `Dock` are used interchangably in both British English and American English.

So I think that even more presets would make it too confusing for the case which is 1000 times more common than this obscure feature

**Comment by matkoniecz** [comment]
> i don't see motorway mapped as footway as a severe issue

well, it seems clear that judgment what is severe data quality issue differs significantly here
